pkg/schedule/schedulers/shuffle_hot_region.go

```diff
@@ -77,6 +115,24 @@ func (s *shuffleHotRegionScheduler) EncodeConfig() ([]byte, error) {
 	return EncodeConfig(s.conf)
 }
 
+func (s *shuffleHotRegionScheduler) ReloadConfig() error {
+	s.conf.Lock()
+	defer s.conf.Unlock()
+	cfgData, err := s.conf.storage.LoadSchedulerConfig(s.GetName())
+	if err != nil {
+		return err
+	}
+	if len(cfgData) == 0 {
+		return nil
+	}
+	newCfg := &shuffleHotRegionSchedulerConfig{}
```

Comment:
When do we need to create a newCfg instead of using the old one?

Reply:
When the scheduling server watches the change of config, it will call `ReloadConfig` to update it.

Reply:
I mean such as `splitBucketScheduler`, the ReloadConfig doesn't create a newCfg.